### ISSUE TITLE
ImportError: DLL load failed while importing _errors

### ISSUE THREAD

**Issue by AlexanderJuestel** [h5py/h5py#2305]
Hello, 

I am using the [rasterio](https://github.com/rasterio/rasterio) package together with the [segysak](https://github.com/trhallam/segysak) package which relies on h5py. When importing these packages as seen below, an `ImportError` is thrown.

To reproduce:

```python
import rasterio
import h5py
``` 

To assist reproducing bugs, please include the following:
 * Operating System: Windows 10
 * Python version: 3.10
 * Where Python was acquired: Anaconda
 * h5py version: 3.9.0
 * HDF5 version (e.g. 1.8.17)
 * The full traceback/stack trace shown (if it appears)

Summary of the h5py configuration
---------------------------------

h5py    3.9.0
HDF5    1.12.2
Python  3.10.12 | packaged by Anaconda, Inc. | (main, Jul  5 2023, 19:01:18) [MSC v.1916 64 bit (AMD64)]
sys.platform    win32
sys.maxsize     9223372036854775807
numpy   1.23.5
cython (built with) 0.29.35
numpy (built against) 1.21.6
HDF5 (built against) 1.12.2

```python
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[1], line 2
      1 import rasterio
----> 2 import h5py

File ~\Anaconda3\envs\pygeomechanical\lib\site-packages\h5py\__init__.py:25
     19 # --- Library setup -----------------------------------------------------------
     20 
     21 # When importing from the root of the unpacked tarball or git checkout,
     22 # Python sees the "h5py" source directory and tries to load it, which fails.
     23 # We tried working around this by using "package_dir" but that breaks Cython.
     24 try:
---> 25     from . import _errors
     26 except ImportError:
     27     import os.path as _op

ImportError: DLL load failed while importing _errors: Die angegebene Prozedur wurde nicht gefunden.
```

**Comment by ajelenak** [comment]
Seems same as #2303.

**Comment by AlexanderJuestel** [comment]
@ajelenak also saw that already. rasterio also depends on GDAL....

**Comment by ajelenak** [comment]
I don't think this is related to h5py itself but some problem in producing binary packages. I don't have this problem on my macOS laptop.

```python
% conda create -n foo h5py rasterio python=3.11
% conda activate foo
(foo) ~ % python
Python 3.11.5 | packaged by conda-forge | (main, Aug 27 2023, 03:35:23) [Clang 15.0.7 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import rasterio
>>> import h5py
>>>
```

**Comment by AlexanderJuestel** [comment]
@ajelenak so you would suggest reporting that in the rasterio repo?

**Comment by ajelenak** [comment]
I suggest this 3 steps:

1. Use _conda_ command to install all packages available from the conda channel you are using. Use _pip_ (from the same conda env.) for any remaining package. 
2. Consider using the conda-forge channel as the default repository.
3. If your problem remains, report it to the maintainers of the conda channel from where the packages came.

**Comment by FmingC** [comment]
I'm having the same problem. Can you tell me how to solve it?

**Comment by GF-Huang** [comment]
Install by `conda`, will cause the same issue. But install by `pip`, it works well.
